### PR TITLE
Merge the Land Unit doctype with the Location doctype

### PR DESCRIPTION
**Pull request by Alchez**
Pull-Request

- [x] Have you followed the guidelines in our Contributing document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
- [x] Have you lint your code locally prior to submission?
- [ ] Have you successfully run tests with your changes locally?
- [x] Does your commit message have an explanation for your changes and why you'd like us to include them?
- [ ] Docs have been added / updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Did you modify the existing test cases? If yes, why?

---

What type of a PR is this? 

- [x] Changes to Existing Features
- [ ] New Feature Submissions
- [ ] Bug Fix
- [x] Breaking Change

--- 

- Motivation and Context (What existing problem does the pull request solve):

The Land Unit and Location DocTypes are essentially the same and refer to the same thing. While working with the Agriculture module using different kinds of Assets, we found that there are potential scenarios for redundancies and/or duplicate entries when working with Land Units (Agri) and Location (Asset).

This PR merges these DocTypes to avoid these problems.

- Screenshots (if applicable, remember, a picture tells a thousand words):

image

### CODE REVIEW

**Comment by Alchez** [comment]
@nabinhait, I was having problems writing the patch for merging the existing Land Unit records. I requested @codingCoffee to help me out with that, so I'll push the patch once we've locked that down.

File: erpnext/agriculture/doctype/crop_cycle/test_crop_cycle.py [hunk 1]
```diff
@@ -3,66 +3,72 @@
...
 		self.assertEqual(check_task_creation(), True)
 		self.assertEqual(check_project_creation(), True)
 
+
 def check_task_creation():
 	all_task_dict = {
 		"Survey and find the aphid locations": {
-			"exp_start_date": frappe.utils.datetime.date(2017,11,21),
-			"exp_end_date": frappe.utils.datetime.date(2017,11,22)
+			"exp_start_date": datetime.date(2017, 11, 21),
```
Comment (codingCoffee) [inline erpnext/agriculture/doctype/crop_cycle/test_crop_cycle.py]: any reason for not using `frappe.utils` ??

**Review by codingCoffee — CHANGES_REQUESTED** [review]

File: erpnext/agriculture/doctype/crop_cycle/test_crop_cycle.py (hunk 1 above)
Comment (Alchez) [reply erpnext/agriculture/doctype/crop_cycle/test_crop_cycle.py]: Not functionally, this was just a sanity edit. :sweat_smile:

File: erpnext/patches/v11_0/merge_land_unit_with_location.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+	frappe.reload_doc('assets', 'doctype', 'location')
+
+	if frappe.db.table_exists('Linked Land Unit') and not frappe.db.table_exists('Linked Location'):
+		rename_doc('DocType', 'Linked Land Unit', 'Linked Location', force=True)
+
+	frappe.reload_doc('assets', 'doctype', 'linked_location')
+
+	# Rename the fields in related doctypes
+	if 'lanked_land_unit' in frappe.db.get_table_columns('Crop Cycle'):
```
[the remark this answers is not in the document]
Comment (Alchez) [reply erpnext/patches/v11_0/merge_land_unit_with_location.py]: Ah, I'll fix that.

**Comment by codingCoffee** [comment]
bump

**Review by nabinhait — APPROVED** [review]

**Comment by harshit-30** [comment]
@Alchez in the above screenshot, you have selected a square to mark some land and the Area field generated some value in it, what is the UOM of that field and also, what is the Use Case of providing an additional field Area UOM in the form?

**Comment by Alchez** [comment]
@harshit-30, OpenStreetMap will default all measurements in square meters. I should have added that default value in the "Area UOM", and it should be read-only, since there's no current provision to easily view conversions.

**Pull request merged**